### PR TITLE
Better safe than sorry: really check only the hostname

### PR DESCRIPTION
**Pull request by ariya**
Thus, avoid matching for "openstreetmap.org" in any other parts of the URL.
https://codeql.github.com/codeql-query-help/javascript/js-incomplete-url-substring-sanitization/

### CODE REVIEW

File: frontend/src/metabase/visualizations/components/LeafletMap.jsx [hunk 1]
```diff
@@ -49,7 +49,7 @@ export default class LeafletMap extends Component {
 
       const mapTileUrl = MetabaseSettings.get("map-tile-server-url");
       const mapTileAttribution =
-        mapTileUrl.indexOf("openstreetmap.org") >= 0
+        new URL(mapTileUrl).host.indexOf("openstreetmap.org") >= 0
```
Comment (paulrosenzweig) [inline frontend/src/metabase/visualizations/components/LeafletMap.jsx:52]: `new URL` will throw if it's not passed a valid URL. Hopefully we're validating that, but it's probably worth adding a try/catch to handle bad strings getting in there.

Also, `.indexOf("openstreetmap.org") >= 0` could be `.includes("openstreetmap.org")`.

Outside of this PR, maybe we should refine this check. You can serve OSM data from other hosts. Maybe letting the user toggle it on/off is best?

**Review by paulrosenzweig — APPROVED** [review]
LGTM!

File: frontend/src/metabase/visualizations/components/LeafletMap.jsx (hunk 1 above)
Comment (ariya) [reply frontend/src/metabase/visualizations/components/LeafletMap.jsx:52]: Good point on adding a try/catch.

As for the(custom) OSM tile URL, I think it's already part of the settings. The above logic is only to explicitly add the attribution, which is the requirement of using OSM tiles from openstreetmap.org (I'm not sure we want this to be toggle-able).

Comment (paulrosenzweig) [reply frontend/src/metabase/visualizations/components/LeafletMap.jsx]: > which is the requirement of using OSM tiles from openstreetmap.org

I think the requirement applies for OSM data regardless of where it's hosted. If I spun up my own tile server using OSM data on `tiles.example.com`, this check wouldn't display the needed message. OSM maintains their [own list](https://wiki.openstreetmap.org/wiki/Tile_servers) of different hosts, so it's likely people swap in others.

**Pull request merged**